components/ActionBox/components/DealActionBox.tsx

```diff
@@ -0,0 +1,265 @@
+import Image from 'next/image';
+import styled from 'styled-components';
+import { useEffect, useState, useMemo } from 'react';
+
+import Info from 'assets/svg/info.svg';
+
+import Connector from 'containers/Connector';
+
+import { Status } from 'components/DealStatus';
+import QuestionMark from 'components/QuestionMark';
+import ConfirmTransactionModal from 'components/ConfirmTransactionModal';
+
+import { FlexDivRowCentered, Tooltip } from '../../common';
+
+import {
+	Container,
+	ErrorNote,
+	ContentContainer,
+	ActionBoxInputLabel,
+	InputContainer,
+	ActionBoxInput,
+	ActionBoxMax,
+	ActionBoxHeaderWrapper,
+	ActionBoxHeader,
+	FlexDivCenterRow,
+	ActionButton,
+} from './commons';
+
+import { TransactionType, TransactionStatus } from 'constants/transactions';
+import { statusToText, swimmingPoolID } from 'constants/pool';
+
+const DealActionBox = ({
+	poolId,
+	dealRedemptionData,
+	purchaseCurrency,
+	transaction: { txType, txState, setTxType, setGasPrice, gasLimitEstimate, onSubmit },
```

Comment:
it might be ok to keep passing the transaction object in here like this but we can also just regrab the data from the Container if we want. I understand that it might be convenient in this case although we usually dont pass the data through like that if they are stored in Containers which is sort of like a global state any component can already access. i think we were already doing this although maybe we want to stop? I will let you decide

Reply:
Done